### PR TITLE
Fix cache updating subqueries that does not have typename

### PR DESCRIPTION
**Pull request by degroote22**
It is a follow up to this PR: https://github.com/zino-app/graphql-flutter/pull/156
I ended up still having the issue where the subquery would override the superquery when there wasn't a typename on the field.

This PR adds tests for this edge case and fixes it.

I'm sorry I didn't put this all on the same PR.

### CODE REVIEW

File: lib/src/cache/in_memory.dart [hunk 1]
```diff
@@ -2,11 +2,25 @@ import 'dart:async';
 import 'dart:collection';
 import 'dart:convert';
 import 'dart:io';
-
 import 'package:path_provider/path_provider.dart';
-
 import 'package:graphql_flutter/src/cache/cache.dart';
 
+void _recursiveAddAll(
```
Comment (HofmannZ) [inline lib/src/cache/in_memory.dart]: Shouldn't this be a helper function inside `utilities/helpers.dart` ?

File: lib/src/cache/in_memory.dart [hunk 2]
```diff
@@ -2,11 +2,25 @@ import 'dart:async';
 import 'dart:collection';
 import 'dart:convert';
 import 'dart:io';
-
```
Comment (HofmannZ) [inline lib/src/cache/in_memory.dart]: We tend to keep Dart packages separate from installed packages.

File: lib/src/cache/in_memory.dart [hunk 3]
```diff
@@ -2,11 +2,25 @@ import 'dart:async';
...
 import 'package:path_provider/path_provider.dart';
-
```
Comment (HofmannZ) [inline lib/src/cache/in_memory.dart]: We also tend to keep project packages separate from dart and installed packages.

**Comment by micimize** [comment]
@HofmannZ moved the helper and broke up the imports, so this is good to go.
